File: Pila.cpp
#include <iostream>
#include <stack>

using namespace std;

// Función para mostrar la pila sin borrar los datos originales
void recorrerPila(stack<int> pila) {
    if (pila.empty()) {
        cout << "La pila esta vacia." << endl;
        return;
    }
    while (!pila.empty()) {
        cout << "[" << pila.top() << "] ";
        pila.pop();
    }
    cout << endl;
}

int main() {
    stack<int> miPila;
    int numero;

    // 1. Insertar (push): El usuario digita los 5 elementos solicitados
    cout << "Implementacion de una Pila (Referencia: image_1a6315.png)" << endl;
    cout << "Por favor, ingrese 5 numeros:" << endl;
    
    for (int i = 0; i < 5; i++) {
        cout << "Digite valor para la posicion " << i + 1 << ": ";
        cin >> numero;
        miPila.push(numero);
    }

    // 2. Mostrar el tope
    cout << "\nEl elemento en el tope es: " << miPila.top() << endl;

    // 3. Recorrer la pila (Estado inicial)
    cout << "Estado actual de la pila: ";
    recorrerPila(miPila);

    // 4. Eliminar (pop) dos elementos consecutivos
    cout << "\nEliminando los dos ultimos elementos ingresados..." << endl;
    for (int i = 0; i < 2; i++) {
        if (!miPila.empty()) {
            cout << "Eliminando (pop): " << miPila.top() << endl;
            miPila.pop();
        }
    }

    // 5. Mostrar el estado final
    cout << "\n--- ESTADO FINAL ---" << endl;
    if (!miPila.empty()) {
        cout << "Nuevo tope: " << miPila.top() << endl;
    }
    cout << "Elementos que quedaron en la pila: ";
    recorrerPila(miPila);

    return 0;
}

File: Pila.py
# Función para recorrer y mostrar la pila sin destruirla
def recorrer_pila(pila):
    if not pila:
        print("La pila está vacía.")
        return
    # Mostramos una copia invertida para simular el orden de la pila (Tope -> Fondo)
    elementos = list(pila)
    elementos.reverse()
    print("Contenido de la pila (Tope -> Fondo):", elementos)

def main():
    pila = []

    # 1. Insertar (push): El usuario digita los 5 elementos
    print("--- Implementación de Pila en Python ---")
    print("Digite los 5 elementos solicitados (ej. 5, 10, 15, 20, 25):")
    
    for i in range(5):
        num = int(input(f"Elemento {i+1}: "))
        pila.append(num)  # append() actúa como push

    # 2. Mostrar el tope
    if pila:
        print(f"\nTope actual: {pila[-1]}") # El índice -1 accede al último elemento

    # 3. Recorrer
    recorrer_pila(pila)

    # 4. Eliminar (pop) dos elementos consecutivos
    print("\nEliminando los dos últimos elementos...")
    for _ in range(2):
        if pila:
            eliminado = pila.pop()  # pop() elimina y devuelve el último
            print(f"Pop: {eliminado}")

    # 5. Mostrar el estado final
    print("\n--- ESTADO FINAL ---")
    if pila:
        print(f"Nuevo tope: {pila[-1]}")
    recorrer_pila(pila)

if __name__ == "__main__":
    main()
